### DISCUSSION TITLE
Applying discounts to product scope

### DISCUSSION THREAD

**Discussion by CWFranklin** [lunarphp/lunar#1037] [Feature Requests]
I would like to be able to display globally applied discounts (like a sitewide sale) in any location that the product is displayed.

Currently the only way of applying a discount is via a cart. It would be good if the Discount Manager could also be used with some sort of `applyToProduct` function in order to display this information.

I have considered utilising a software cart to apply the discount to products but this would be very taxing for a site with any sort of category page. I have also considered manually applying percentage discounts but this would mean duplicating some existing site code and it would not be updated along with any other fixes to the core site code.

**Comment by alecritson** [comment]
Hi @CWFranklin If the discount is site wide and applies to any product could you not just get the discount from the database?

If the discount is for specific products, couldn't you just get all discounts that are restricted to that product?

**Comment by CWFranklin** [comment]
Hey Alec. Yeah, getting it from the database was one of my considered options. While I figured it was unlikely that it would change, I was hoping to avoid duplicating the percentage discount logic. Much like the apply code checks against cart lines, I thought an applyToPurchasable function could check criteria against products. The extended code around the percentage calculations also check against customer group, etc.

**Comment by wychoong** [comment]
would like to add on that this would be useful to show the discounts for the products for example

in product list
![image](https://github.com/lunarphp/lunar/assets/67364036/594341e2-0aa5-4ba5-8ffe-17a7bb9d5de3)

in product page
![image](https://github.com/lunarphp/lunar/assets/67364036/cbac84ed-fe99-4a26-8532-e16e0b09fd3c)

as mentioned currently the discount manager only calculate for products in cart.
to achieve for this product level discount we will need to take certain considerations:
- Cart info (currency, coupon code, items in cart) due to rules like `minimum cart amount`, `buy x get y`
- show the lowest possible discount value
